--- ipp-crypto/include/ippcp.h	2019-10-16 18:12:26.157669380 +0800
+++ inc/ippcp.h	2019-10-28 23:53:28.871586537 +0800
@@ -49,6 +49,11 @@
 #define IPPCP_H__
 
 
+#ifndef _SGX_IPPCP_H_
+#include "sgx_ippcp.h"
+#endif
+
+
 #ifndef IPPCPDEFS_H__
   #include "ippcpdefs.h"
 #endif
@@ -70,11 +75,6 @@
   #endif
 #endif
 
-#pragma warning(push)
-#ifdef _MSC_VER
-#pragma warning(disable : 4100) // for MSVC, unreferenced param
-#endif
-
 /* /////////////////////////////////////////////////////////////////////////////
 //  Name:       ippcpGetLibVersion
 //  Purpose:    getting of the library version
@@ -1331,7 +1331,6 @@
 IPPAPI(IppStatus, ippsGFpECESGetBuffersSize_SM2, (int* pPublicKeySize,
                         int* pMaximumTagSize, const IppsECESState_SM2* pState))
 
-#pragma warning(pop)
 #ifdef  __cplusplus
 }
 #endif
